src/services/ChallengeService.js

```diff
@@ -956,6 +956,12 @@ async function createChallenge (currentUser, challenge, userToken) {
       }
     }
   }
+  if (challenge.legacy.selfService) {
+    if (!challenge.metadata) {
+      challenge.metadata = {}
```

Comment:
I think the challenge metadata is normally an array, and we shouldn't need `.items()` in line 963

Reply:
Agreed